Add support for discrete axis values and axis source

emersion opened the pull request:
Test plan: `env WAYLAND_DEBUG=1 weston-clickdot` and check that `axis_discrete` and `axis_source` events are sent. Note that not all hardware supports this.

cc @Ongy @Timidger: this requires changes in your libraries and compositors.

emersion commented on backend/x11/input_device.c:
```diff
@@ -73,13 +73,14 @@ void handle_x11_input_event(struct wlr_x11_backend *x11,
 				.source = WLR_AXIS_SOURCE_WHEEL,
 				.orientation = WLR_AXIS_ORIENTATION_VERTICAL,
-				.delta = delta,
+				.delta = delta_discrete * 15,
```
>We should probably note that '15' is just a typical value that libinput returns for scrolls.

Yeah, good idea. Also changed the value to 10 to match xwayland's behaviour.

ascent12 replied:
>Also changed the value to 10 to match xwayland's behaviour.

That was the old behaviour. It's definitely 15 now.

emersion replied:
Ah, changed it back.

ascent12 commented:
How much extra work is axis_source on top of this? If we're going to break the API here, it'll probably be less hassle for our users to do both at once.

emersion commented:
>How much extra work is axis_source on top of this? If we're going to break the API here, it'll probably be less hassle for our users to do both at once.

Yeah, that's a good idea, implemented it.

Note that we don't send frame events correctly because we split events that scroll on multiple axes in the libinput backend.

ascent12 commented on types/seat/wlr_seat_pointer.c:
```diff
@@ -241,6 +241,9 @@ void wlr_seat_pointer_send_axis(struct wlr_seat *wlr_seat, uint32_t time,
 		uint32_t version = wl_resource_get_version(resource);
 
+		if (version >= WL_POINTER_AXIS_SOURCE_SINCE_VERSION) {
+			wl_pointer_send_axis_source(resource, source);
```
Is our axis source enum guaranteed to be compatible with the wayland one?

emersion replied:
Yeah, just like the orientation one.
